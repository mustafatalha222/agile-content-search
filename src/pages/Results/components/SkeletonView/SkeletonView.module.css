.skeletonList {
  margin: 2rem 11vw;
  list-style-type: none;
  display: flex;
  flex-direction: column;
  gap: 1.5rem;
}

@media (max-width: 768px) {
  .skeletonList {
    margin: 2rem 2vw;
  }
}
