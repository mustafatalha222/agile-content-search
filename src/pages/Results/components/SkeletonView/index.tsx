import { memo } from 'react'
import styles from './SkeletonView.module.css'
import Skeleton from '../../../../components/Skeleton/Skeleton'

function SkeletonView() {
  return (
    <ul className={styles.skeletonList}>
      {[...Array(5)].map((_, idx) => (
        <li className={styles.skeletonListItem} key={idx}>
          <Skeleton height={12} width={200} />
          <Skeleton height={15} width={300} />
          <Skeleton height={20} width="80%" />
        </li>
      ))}
    </ul>
  )
}

export default memo(SkeletonView)
