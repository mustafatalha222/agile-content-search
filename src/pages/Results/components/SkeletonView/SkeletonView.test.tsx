import { render } from '@testing-library/react'
import SkeletonView from '.'

describe('SkeletonView Component', () => {
  it('should render and match snapshot', () => {
    const { asFragment } = render(<SkeletonView />)
    expect(asFragment()).toMatchSnapshot()
  })
})
