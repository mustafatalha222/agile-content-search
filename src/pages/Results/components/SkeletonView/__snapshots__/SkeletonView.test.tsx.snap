// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonView Component should render and match snapshot 1`] = `
<DocumentFragment>
  <ul
    class="skeletonList"
  >
    <li
      class="skeletonListItem"
    >
      <div
        class="skeleton block"
        data-testid="skeleton"
        style="height: 12px; width: 200px;"
      />
      <div
        class="skeleton block"
        data-testid="skeleton"
        style="height: 15px; width: 300px;"
      />
      <div
        class="skeleton block"
        data-testid="skeleton"
        style="height: 20px; width: 80%;"
      />
    </li>
    <li
      class="skeletonListItem"
    >
      <div
        class="skeleton block"
        data-testid="skeleton"
        style="height: 12px; width: 200px;"
      />
      <div
        class="skeleton block"
        data-testid="skeleton"
        style="height: 15px; width: 300px;"
      />
      <div
        class="skeleton block"
        data-testid="skeleton"
        style="height: 20px; width: 80%;"
      />
    </li>
    <li
      class="skeletonListItem"
    >
      <div
        class="skeleton block"
        data-testid="skeleton"
        style="height: 12px; width: 200px;"
      />
      <div
        class="skeleton block"
        data-testid="skeleton"
        style="height: 15px; width: 300px;"
      />
      <div
        class="skeleton block"
        data-testid="skeleton"
        style="height: 20px; width: 80%;"
      />
    </li>
    <li
      class="skeletonListItem"
    >
      <div
        class="skeleton block"
        data-testid="skeleton"
        style="height: 12px; width: 200px;"
      />
      <div
        class="skeleton block"
        data-testid="skeleton"
        style="height: 15px; width: 300px;"
      />
      <div
        class="skeleton block"
        data-testid="skeleton"
        style="height: 20px; width: 80%;"
      />
    </li>
    <li
      class="skeletonListItem"
    >
      <div
        class="skeleton block"
        data-testid="skeleton"
        style="height: 12px; width: 200px;"
      />
      <div
        class="skeleton block"
        data-testid="skeleton"
        style="height: 15px; width: 300px;"
      />
      <div
        class="skeleton block"
        data-testid="skeleton"
        style="height: 20px; width: 80%;"
      />
    </li>
  </ul>
</DocumentFragment>
`;
